csharp/OnnxRuntime.CSharp.proj

```diff
@@ -19,19 +19,19 @@ CMake creates a target to this project
     <Message Importance="High" Text="Restoring NuGet packages for CSharp projects..." />
     <MSBuild Projects="src\Microsoft.ML.OnnxRuntime\Microsoft.ML.OnnxRuntime.csproj"
              Targets="Restore" 
-             Properties="Platform=AnyCPU;RestoreConfigFile=$(MSBuildThisFileDirectory)\NuGet.CSharp.config;MSBuildWarningsAsMessages=NU1503;RestoreIgnoreFailedSource=true" 
+             Properties="Platform=AnyCPU;RestorePackagesPath=$(MSBuildThisFileDirectory)\packages;RestoreConfigFile=$(MSBuildThisFileDirectory)\NuGet.CSharp.config;MSBuildWarningsAsMessages=NU1503;RestoreIgnoreFailedSource=true" 
```

Comment:
An easier way to accomplish this is to set this property in a `Directory.Build.props` file. That way it is only set in 1 spot, when you add a new .csproj you don't need to add it here, and it is set for all targets that are run (not just `Restore`).

We do it that way in ML.NET here: [https://github.com/dotnet/machinelearning/blob/365ccf292789ba1f14d80b13e44daaf951517fff/Directory.Build.props#L53](https://github.com/dotnet/machinelearning/blob/365ccf292789ba1f14d80b13e44daaf951517fff/Directory.Build.props#L53)

Reply:
Just in case you didn't know - Directory.Build.props files are automatically imported to every project under that folder. So if you add the file to the `csharp` directory, all .csproj files under that directory will inherit those properties.

---
In reply to: [313892277](https://github.com/microsoft/onnxruntime/pull/1605#discussion_r313892277) [](ancestors = 313892277)